tests/language-feature/extensions/extension-override.slang

```diff
@@ -0,0 +1,65 @@
+// Test that the override behavior around extensions and generic extensions works as expected.
```

Comment:
I think we need a separate diagnostic to test `Diagnostics::invalidExtensionOnInterface` to ensure this functionality does not break in the future.

Reply:
Done.